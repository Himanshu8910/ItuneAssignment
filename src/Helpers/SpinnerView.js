import React from 'react';
import { View, Modal, ActivityIndicator, Text, StyleSheet } from 'react-native';

const SpinnerView = (props) => {
    return (
        <Modal
            animationIn={'zoomInDown'}
            animationOut={'zoomOutUp'}
            backdropOpacity={0}
            activeOpacity={0.5}
            animationInTiming={5000}
            animationOutTiming={5000}
            backdropTransitionInTiming={5000}
            backdropTransitionOutTiming={5000}
            transparent={true}
            style={styles.container}
        >
            <View style={styles.loaderViewParentStyle}>
                <View style={styles.loaderViewStyle}>
                    <ActivityIndicator size="large" color="black" />
                    <Text style={styles.loaderTextStyle}>{props.text}</Text>
                </View>
            </View>
        </Modal>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    loaderViewParentStyle: {
        flex: 1,
        backgroundColor: 'rgba(0, 0, 0,0.3)',
        alignItems: 'center',
        justifyContent: 'center'
    },

    loaderViewStyle: {
        height: 110,
        width: 170,
        backgroundColor: 'white',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 6,
    },
    loaderTextStyle: {
        marginTop: 14
    }

});


export default SpinnerView;