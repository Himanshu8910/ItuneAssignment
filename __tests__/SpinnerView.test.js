import React from 'react';
import renderer from 'react-test-renderer';

import SpinnerView from '../src/Helpers/SpinnerView';

test('renders correctly', () => {
  const tree = renderer.create(<SpinnerView />).toJSON();
  expect(tree).toMatchSnapshot();
});
