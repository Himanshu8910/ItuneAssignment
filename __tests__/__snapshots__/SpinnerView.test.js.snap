// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  activeOpacity={0.5}
  animationIn="zoomInDown"
  animationInTiming={5000}
  animationOut="zoomOutUp"
  animationOutTiming={5000}
  backdropOpacity={0}
  backdropTransitionInTiming={5000}
  backdropTransitionOutTiming={5000}
  hardwareAccelerated={false}
  style={
    Object {
      "flex": 1,
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0,0.3)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 6,
          "height": 110,
          "justifyContent": "center",
          "width": 170,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="black"
        hidesWhenStopped={true}
        size="large"
      />
      <Text
        style={
          Object {
            "marginTop": 14,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
